### DISCUSSION TITLE
Could we add double square brackets for dynamic routes?

### DISCUSSION THREAD

**Discussion by dande-sun** [withastro/roadmap#861] [Proposal]
### Body

# Summary
 Add double square brackets param for dynamic routes, make the param is optional in the double square brackets.

# Background & Motivation
We need match URLs path:
`/request-demo`
`/a/request-demo`
`/b/request-demo`
now we have two same files:
`/[slug]/request-demo.astro`
`/request-demo.astro`
if slug is optional match, only need one file.
`/[[slug]]/request-demo.astro`
Also the [...slug] too, 

# Goals
`/[slug]/request-demo.astro` match:
/a/request-demo 
/b/request-demo

`/[[slug]]/request-demo.astro` match:
/request-demo
/a/request-demo
/b/request-demo

`/[...slug]/request-demo.astro` match:
/a/request-demo 
/a/b/request-demo

`/[[...slug]]/request-demo.astro` match:
/request-demo
/a/request-demo
/a/b/request-demo

**Comment by delucis** [comment]
The `[...slug]` format already includes support for `slug` being undefined, i.e. optional, so I think does what you need?

**Reply by dande-sun** [reply]
there have a page need match URL `/request-demo`, `/a/request-demo` and `/b/request-demo`, and another page only match url `/a/demo` and `/b/demo`.

**Reply by delucis** [reply]
I think that’s possible?

- `/[...slug]/request-demo.astro` would match `/request-demo`, `/a/request-demo` and `/b/request-demo`
- `/[slug]/demo.astro` would match `/a/demo/` and `/b/demo/`
